Modules/CalcOffence-3_0.lua

```diff
@@ -1847,9 +1847,30 @@ function calcs.offence(env, actor)
 			local baseVal = calcAilmentDamage("Shock", sourceHitDmg, sourceCritDmg)
 			if baseVal > 0 then
 				skillFlags.shock = true
-				output.ShockDurationMod = 1 + modDB:Sum("INC", cfg, "EnemyShockDuration") / 100 + enemyDB:Sum("INC", nil, "SelfShockDuration") / 100
+
+		        local baseShockDur = 2
+		        local incDur = 1 + modDB:Sum("INC", cfg, "EnemyShockDuration") / 100 + enemyDB:Sum("INC", nil, "SelfShockDuration") / 100
```

Comment:
Aren't enemy modifiers and "self" modifiers multiplicative with each other? E.g. `20% increased shock duration` and `Shocks last 20% longer` (from the enemy POV) would result in `total duration = 1.2 * 1.2` or am I misunderstanding where the `SelfShockDuration` comes from? Can you give an example when enemies have that modifier?

Reply:
Only thing I could find was on Conductivity curse https://pathofexile.gamepedia.com/Conductivity
Hypothermia does something similar with it's quality modifier. It is also listed as SelfFreezeDuration in PoB.
SelfFreezeDuration and EnemyFreezeDuration are summed up in the offense calculation, not multiplied together. I think the difference would be if the enemy was also under the effect of Temporal Chains, which is a multiplier on the ailment duration (by nature of making time expire slower)

Reply:
It could also  be that PoB is wrong and they *should* be multiplied :)

Reply:
If this is indeed applied by Conductivity then the calculation is correct. 

I just thought that poe treats these things the same way it treats damage (i.e. damage dealt is multiplicative with damage taken) but looking at Maligaros Restraint it seems like durations are indeed additive.